src/attributes/components/AttributeDetails/NumericUnits.tsx

```diff
@@ -0,0 +1,189 @@
+import { AttributePageFormData } from "@saleor/attributes/components/AttributePage";
+import ControlledCheckbox from "@saleor/components/ControlledCheckbox";
+import SingleSelectField from "@saleor/components/SingleSelectField";
+import { UseFormResult } from "@saleor/hooks/useForm";
+import { commonMessages } from "@saleor/intl";
+import { makeStyles } from "@saleor/theme";
+import { MeasurementUnitsEnum } from "@saleor/types/globalTypes";
+import React, { useEffect, useMemo, useState } from "react";
+import { useIntl } from "react-intl";
+
+import * as M from "./messages";
+import {
+  getUnitChoices,
+  unitMapping,
+  UnitSystem,
+  unitSystemChoices,
+  UnitType,
+  unitTypeChoices
+} from "./utils";
+
+const useStyles = makeStyles(
+  theme => ({
+    unitsRow: {
+      columnGap: theme.spacing(2),
+      display: "flex",
+      [theme.breakpoints.down("sm")]: {
+        flexFlow: "wrap",
+        rowGap: theme.spacing(3)
+      }
+    },
+    hr: {
+      border: "none",
+      borderTop: `1px solid ${theme.palette.divider}`,
+      height: 0,
+      margin: "0.5rem 0",
+      width: "100%"
+    }
+  }),
+  { name: "NumericUnits" }
+);
+
+interface UnitData {
+  unit?: MeasurementUnitsEnum;
+  system?: UnitSystem;
+  type?: UnitType;
+}
+
+interface NumericUnitsProps
+  extends Pick<
+    UseFormResult<AttributePageFormData>,
+    "set" | "setError" | "data" | "errors" | "clearErrors"
+  > {
+  disabled: boolean;
+}
+
+export const NumericUnits: React.FC<NumericUnitsProps> = ({
+  data,
+  disabled,
+  errors,
+  set,
+  setError,
+  clearErrors
+}) => {
+  const { formatMessage } = useIntl();
+  const classes = useStyles();
+  const [unitData, setUnitData] = useState<UnitData>({
+    unit: data.unit
+  });
+
+  const { unit, system, type } = unitData;
+  const errorProps = {
+    error: !!errors.unit,
+    hint: formatMessage(commonMessages.requiredField)
+  };
+  const [typeChoices, systemChoices, unitChoices] = useMemo(
+    () => [
+      unitTypeChoices.map(choice => ({
+        ...choice,
+        label: formatMessage(choice.label)
+      })),
+      unitSystemChoices.map(choice => ({
+        ...choice,
+        label: formatMessage(choice.label)
+      })),
+      getUnitChoices(formatMessage)
+    ],
+    []
+  );
+
+  useEffect(() => set({ unit }), [unit]);
+
+  useEffect(() => {
+    if (data.unit) {
+      const selectInitialUnitData = () => {
+        const initialData: UnitData = { unit: data.unit };
+
+        Object.entries(unitChoices).some(([system, types]) => {
+          const systemMatch = Object.entries(types).find(([type, units]) => {
```

Comment:
```suggestion
          const systemMatch = Object.entries(types).some(([type, units]) => {
```

Reply:
Was sure I've pushed that change as well 🙇